File: q6/plot.R
library(ggplot2)

pdf("densprofile.pdf", width=7, height=6)

waterDens <- read.csv('water.txt')
tailDens  <- read.csv('tail.txt')
glycerolDens  <- read.csv('glycerol.txt')
headgroupDens <- read.csv('headgroup.txt')
ionDens <- read.csv('ion.txt')

par(mar = c(6, 6, 6, 6))

plot(waterDens[,1], waterDens[,2],xlim = c(-33,33), xlab=expression( paste("z ( ",  ring(A),  ") ") ), ylab=expression(paste("density ( mol / ", ring(A)^3,")")), col="blue", ylim=c(0, 0.012), type="l")
lines(ionDens[,1], ionDens[,2], col="orange")
lines(headgroupDens[,1], headgroupDens[,2], col="red")
lines(glycerolDens[,1], glycerolDens[,2], col="green")
lines(tailDens[,1], tailDens[,2], col="black")
legend(x="topleft", bty = "n", c('water', 'ions', 'headgroups', 'glycerol', 'tails'), col=c('blue', 'orange', 'red', 'green', 'black'),  pch=c("-", "-", "-", "-"))

#axis(side = 2,
#     ## Rotate the labels.
#     las = 2,
#     ## Adjust the label position.
#     mgp = c(2, 0.75, 0))

dev.off()

File: q6/plot.py
import numpy as np
import matplotlib.pyplot as plt

#glycerol.txt
#headgroup.txt
#ion.txt
#tail.txt
#water.txt

dens_glycerol=np.genfromtxt('glycerol.txt', delimiter=',')
dens_headgroup=np.genfromtxt('headgroup.txt', delimiter=',')
dens_ion=np.genfromtxt('ion.txt', delimiter=',')
dens_tail=np.genfromtxt('tail.txt', delimiter=',')
dens_water=np.genfromtxt('water.txt', delimiter=',')

fig = plt.gcf()
fig.set_size_inches(3.25, 3.25)
fig.subplots_adjust(bottom=0.15, left=0.25, top=0.8)

ax=plt.axes()

plt.plot(dens_glycerol[:,0], dens_glycerol[:,1], '-', label='glycerol')
plt.plot(dens_headgroup[:,0], dens_headgroup[:,1], '-', label='headgroup')
plt.plot(dens_tail[:,0], dens_tail[:,1], '-', label='tails')
plt.plot(dens_ion[:,0], dens_ion[:,1], '-', label='ions')
plt.plot(dens_water[:,0], dens_water[:,1], '-', label='water')
plt.legend(bbox_to_anchor=(0., 1.02, 1., .102), loc=3,
           ncol=2, mode="expand", borderaxespad=0., frameon=False)

ax.set(xlim=(-33,33), ylim=(0, 0.01), xlabel=r'Z $(\AA)$', ylabel=r'density (mol/$\AA^3$)')

plt.show()
plt.savefig('q6.pdf')
